src/hoi4_world/military/convoy_distributor.cpp

```diff
@@ -0,0 +1,54 @@
+#include "src/hoi4_world/military/convoy_distributor.h"
+
+namespace hoi4
+{
+
+void ConvoyDistributor::CalculateStateWeights(const vic3::World& source_world)
+{
+   const auto& states = source_world.GetStates();
+   const auto& buildings = source_world.GetBuildings();
+
+   vic3_total_ = 0;
+   vic3_map_ = {};
+   for (const auto [id, state]: states)
+   {
+      const auto building = buildings.GetBuildingInState(id, vic3::BuildingType::Port);
+      if (!building)
+      {
+         continue;
```

Comment:
You should have a test case (or include in an existing case) states with no ports.

Reply:
Done.